### PR TITLE
Make RPC client try more

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: pkg/morph/client/client.go [hunk 1]
```diff
@@ -493,28 +451,26 @@ func (c *Client) TxHalt(h util.Uint256) (res bool, err error) {
 
 // TxHeight returns true if transaction has been successfully executed and persisted.
 func (c *Client) TxHeight(h util.Uint256) (res uint32, err error) {
-	c.switchLock.RLock()
-	defer c.switchLock.RUnlock()
+	var conn = c.conn.Load()
```
Comment (carpawell) [inline pkg/morph/client/client.go:454]: so many `var`s at the first line of the func...

File: pkg/morph/client/constructor.go [hunk 2]
```diff
@@ -106,101 +106,90 @@ func New(key *keys.PrivateKey, opts ...Option) (*Client, error) {
...
-
-			curNotifyChan: make(chan *state.ContainedNotificationEvent),
-			curBlockChan:  make(chan *block.Block),
-			curNotaryChan: make(chan *result.NotaryRequestEvent),
-		},
-	}
-
-	var err error
-	var act *actor.Actor
+	var (
```
Comment (carpawell) [inline pkg/morph/client/constructor.go:109]: even here? more parentheses, more nesting, more lines

**Review by carpawell — APPROVED** [review]
I do not mind the code but not sure I understand why this PR is so big, "Reconnect indefinitely." could be solved in a single line change while the other functional changes just made this client less flexible (this client is used in SN, IR and even used to be used in external client or maybe still).

**Comment by roman-khimov** [comment]
The other change here is that the client will fail requests easily instead of blocking when it's disconnected. It's harder to predict the effect, but it seems to be more fair (associated requests will fail quickly instead of waiting for unknown time).

File: pkg/morph/client/client.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/morph/client/client.go:454]: Love Pascal.

File: pkg/morph/client/constructor.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/morph/client/constructor.go:109]: I can make add even more of them into this `var ()` block for you. But hey, we're defining some variables, `var ()` is exactly for that and it's preferred over a series of `var ... var ... var ...`.

**Review by cthulhu-rider — APPROVED** [review]
woth to be changeloged

File: pkg/morph/client/constructor.go (hunk 2 above)
Comment (carpawell) [reply pkg/morph/client/constructor.go:109]: as always, i prefer `:=`

**Pull request merged**